sound-design.MD

```diff
@@ -0,0 +1,42 @@
+# MRE Sound API proposal
+
+This is node.js side only. It doesn't actually do anything yet.
+
+## Required functionality
+* preload sounds (as looping or one-shot)
+* play preloaded sounds
+* load and play sounds (Basically preload, instantly play, unload upon completion)
+* pause, resume, 
+* stop sounds 
+* set the sound state (to change volume, pitch, and where possible seek)
+
+## Design thoughts
+
+(Feel free to comment on each item separately)
+
+I'm using assetmanager to for the preloading sound to ask the client to load a sound resource, and return a handle. This is therefore are not instant, but return a promise.
+
+All sounds play on an actor, and the play calls therefore happen on the actor.
+
+PlaySound are played by passing in a sound resource ID, and instantly return a sound instance ID.
```

Comment:
>instantly return [](start = 60, length = 16)

Via a ForwardPromise, like Actors?

Reply:
that's not originally what I was thinking, but I agree that forwardpromise is the right thing

---
In reply to: [256944035](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/191#discussion_r256944035) [](ancestors = 256944035)